src/injectors/iframe/frame-state.ts

```diff
@@ -7,11 +7,22 @@ export class FrameHandler {
     this.frameElement = frameElement;
   }
 
-  public async waitForFrameLoad(timeout?: number) {
-    return new Promise<void>((resolve, reject) => {
-      if (!timeout) {
-        timeout = 60 * 1000; // 1 min for default timeout;
+  public async waitForFrameLoad() {
+    return Promise.all([this.checkStatus(), this.checkFrameLoad()]);
+  }
+
+  private checkStatus() {
+    return fetch(this.frameElement.src, { method: 'HEAD' }).then(response => {
```

Comment:
When the `src` is not found (or returned 500) the page is loaded by the browser.
So, reading the status code (using method HEAD) is possible to check if we are showing an error page and handle it.